core/src/test/scala/org/apache/spark/sql/delta/SnapshotManagementSuite.scala

```diff
@@ -56,121 +68,208 @@ class SnapshotManagementSuite extends QueryTest with SQLTestUtils with SharedSpa
   }
 
   testQuietly("recover from a corrupt checkpoint: previous checkpoint doesn't exist") {
-    withTempDir { tempDir =>
-      val path = tempDir.getCanonicalPath
-      spark.range(10).write.format("delta").save(path)
-      val deltaLog = DeltaLog.forTable(spark, path)
-      deltaLog.checkpoint()
-
-      // We have different code paths for empty and non-empty checkpoints
-      for (testEmptyCheckpoint <- Seq(true, false)) {
-        makeCorruptCheckpointFile(path, checkpointVersion = 0, shouldBeEmpty = testEmptyCheckpoint)
-        DeltaLog.clearCache()
-        // Checkpoint 0 is corrupted. Verify that we can still create the snapshot using
-        // existing json files.
-        DeltaLog.forTable(spark, path).snapshot
+    for (useMultipart <- Seq(true, false)) {
+      // Test the first part and not the first part
+      val partsToCorrupt = if (useMultipart) {
+        Some(1) :: Some(2) :: Nil
+      } else {
+        None :: Nil
+      }
+      val maxActions = if (useMultipart) 1 else 10000
+      withSQLConf(DeltaSQLConf.DELTA_CHECKPOINT_MAX_ACTIONS_PER_FILE.key -> maxActions.toString) {
```

Comment:
To minimize the changes in this file, can you try creating a test utility method

```
testWithAndWithoutMultipartCheckpoint(testname) { (isMultiParth, multiPartSize) =>
   // existing test code with some updates needed for 
}
```

```
def testWithAndWithoutMultipartCheckpoint(name)(f: (bool, int) -> Unit)) = {
   // here you can have the setup of multi-part or single part.
}
```

Reply:
Yeah i'll try it

Reply:
Done, let me know how it looks